src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -670,6 +670,18 @@ def _get_registry_details(cmd, app: "ContainerApp", source):
     )
 
 
+def _validate_containerapp_name(name):
+    is_valid = True
+    is_valid = is_valid and name.lower() == name
+    is_valid = is_valid and len(name) <= 32
+    is_valid = is_valid and '--' not in name
```

Comment:
Hmm instead of replacing these unsupported characters with spaces? Why do we not throw an error to the user to not these characters?

Reply:
Are you referring to the line with `name = name.replace('-', '')`? 
Since the dash ("-") is an allowed character, and the only other allowed characters are alphanumeric ones, we eliminate the dashes from the name string so we can check if the rest of the name is alphanumeric by calling `name.isalnum()`. Note that we're just changing a local variable here so we're not actually changing what the containerapp name will be 

@haroonf correct me if I'm wrong on that

Reply:
I see - I am wondering why don't we just error out if there is a - & throw an error with the message on what the expected format is?

Reply:
Can we update help text for this parameter to display what the supported format it? Don't see that change.